manifests/qualys_eus_reporting.pp

```diff
@@ -0,0 +1,81 @@
+# @summary Allow qualys to sudo to root and add wrapper for subscription-manager to lie about enabled repos.
+#   See SVC-12072 for details
+#
+# @param enabled
+#   Boolean to enable/disable wrapper for subscription-manager
+#
+# @param repos
+#   Array of hashes containing repos to add to the qualys_eus_reporting.sh script
+#
+# @example
+#   include profile_audit::qualys_eus_reporting
+class profile_audit::qualys_eus_reporting (
+  Boolean     $enabled,
+  Array[Hash] $repos,
+){
+
+  # Alias that gets added to roots .bashrc
+  $bash_alias = 'function subscription-manager { /root/qualys_eus_reporting.sh "$@"; } #qualys_EUS_fake'
+
+  if ($enabled) {
+    $sudo_ensure_parm = 'present'
+
+    include rhsm
+
+    if ( $::rhsm::manage_repos ) {
+
+      # Using rhsm to manage repos, remove qualys repo lying if present
+      $script_ensure_parm = 'absent'
+
+      exec { 'remove_qualys_EUS_alias':
+        path    => '/bin:/usr/bin',
+        command => "sed -i \'\\|${bash_alias}|d\' /root/.bashrc",
+        onlyif  => "grep \'${bash_alias}\' /root/.bashrc",
+      }
+
+    } else {
+      # Not using rhsm, need to lie about repos
+      $script_ensure_parm = 'present'
+
+      exec { 'add_qualys_EUS_alias':
+        path    => '/bin:/usr/bin',
+        command => "sed -i \'\$a${bash_alias}\' /root/.bashrc",
+        unless  => "grep \'${bash_alias}\' /root/.bashrc",
+      }
+    }
+
+    pam_access::entry { 'Allow sudo for qualys':
+      user       => 'qualys',
+      origin     => 'LOCAL',
+      permission => '+',
+      position   => '-1',
+    }
```

Comment:
I think the `sudo` functionality for Qualys should be moved to the `qualys.pp` file/class. It makes sense to me that Qualys should be able to sudo by default, regardless of whether we are enabling EUS. 

If a particular deployment does not want the default sudo access for Qualys, they can simply unset the hiera value for the sudoers file that qualys would use.

e.g. I could see admin/provisioning type nodes in a cluster (log collectors, Puppet, xCAT, etc) possibly running the latest RHEL 8.x, while actual compute node type nodes being locked to a specific EUS version. In that case Qualys would distinguish between the types of nodes and will need to sudo on them regardless of EUS being setup.

Reply:
But I think that Qualys will be set to try to use sudo based on the definition in the Qualys config for that subnet. For nodes on these subnets that are not flagged in Puppet as EUS this may generate a bunch of email traffic about Qualys trying to sudeo and not having permission to.

Reply:
Oh yes I see what you're saying. Qualys elevated scans would be set on the subnet level, but as-is puppet wasn't fine grained enough to allow sudo separate from enabling the EUS fake reporting.

Reply:
This has been done.